src/platform/packages/shared/kbn-analytics/src/performance/performance_tracker.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { v4 as uuidv4 } from 'uuid';
+import { groupBy } from 'lodash';
+
+export const PERFORMANCE_TRACKER_MARKS = {
+  PRE_RENDER: 'preRender',
+  RENDER_START: 'renderStart',
+  RENDER_COMPLETE: 'renderComplete',
+} as const;
+export type PerformanceTrackerMarks =
+  (typeof PERFORMANCE_TRACKER_MARKS)[keyof typeof PERFORMANCE_TRACKER_MARKS];
+
+export const PERFORMANCE_TRACKER_MEASURES = {
+  PRE_RENDER_DURATION: 'preRenderDuration',
+  RENDER_DURATION: 'renderDuration',
+} as const;
+export type PerformanceTrackerMeasures =
+  (typeof PERFORMANCE_TRACKER_MEASURES)[keyof typeof PERFORMANCE_TRACKER_MEASURES];
+
+interface PerformanceTrackerOptions {
+  type: string;
+  instance: string;
+}
+
+export const createPerformanceTracker = ({ type, instance }: PerformanceTrackerOptions) => {
+  const id = uuidv4();
+  const createMarkName = (name: string) => `${type}:${instance}:${name}`;
+
```

Comment:
We need to specify that the performance.mark should only be used within our EBT testing environment and not in production.

Reply:
Is that something we should just mention in a comment for now? In this PR we don't create a new call to collect performance metrics but add additional fields to `dashboard_loaded`. Or do you mean we should just add additional data to `dashboard_loaded` in the testing environment and otherwise leave it as is?

Reply:
I added checks to call anything related to `performance.*` only if the environment is not production in [dcb7f42](https://github.com/elastic/kibana/pull/216330/commits/dcb7f42acafc9c73e36a5d0ee43914d9b778905d).